tests/TestHarness/launch_transaction_generators.py

```diff
@@ -12,16 +12,19 @@
 
 from .testUtils import Utils
 from pathlib import Path
+from re import split
 
 Print = Utils.Print
 
 class TpsTrxGensConfig:
 
-    def __init__(self, targetTps: int, tpsLimitPerGenerator: int):
+    def __init__(self, targetTps: int, tpsLimitPerGenerator: int, connectionPairList: list):
         self.targetTps: int = targetTps
         self.tpsLimitPerGenerator: int = tpsLimitPerGenerator
-
+        self.connectionPairList = connectionPairList
+        self.numConnectionPairs = int(len(self.connectionPairList)/2)
         self.numGenerators = math.ceil(self.targetTps / self.tpsLimitPerGenerator)
+        self.numGenerators = math.ceil(self.numGenerators/self.numConnectionPairs) * self.numConnectionPairs
```

Comment:
I think what you probably want here is the least common multiple, which newer versions of Python (>3.8 maybe?) support, but since we're bringing in `numpy` already, might as well use `numpy.lcm(self.numGenerators, self.numConnectionPairs)`

Reply:
Wouldn't LCM be giving us bad results?
Take for example a run with 10,000 TPS and 1,500 max trx per gen, where we pass along 4 endpoint pairs.
The old math would put us at 7 generators.
Doing LCM of (7,4) would give us 28 generators, 7 per pair.
Doing the math I have here would give us 8 generators, 2 per pair.

Reply:
`LCM(math.ceil(self.numGenerators/self.numConnectionPairs), self.numConnectionPairs)`
is what we want

Reply:
Nvm. What you have works.